tsl/src/remote/connection.c

```diff
@@ -858,6 +858,14 @@ remote_connection_get_status(const TSConnection *conn)
 const char *
 remote_connection_node_name(const TSConnection *conn)
 {
+#ifndef NDEBUG
```

Comment:
Why don't we use `#ifdef DEBUG` like the rest of the code? I guess at this point in time `#ifdef DEBUG` should be identical to `#ifndef NDEBUG`. Is this always going to be the case? Just wondering about consistency.

Reply:
`NDEBUG` is in the C standard, and I'm not sure where the `DEBUG` comes from, so just using the more standard thing.

Reply:
We explicitly introduced DEBUG in our codebase I believe, probably many years ago. `NDEBUG` is standard when it exists, but we don't explicitly define it. Currently, `NDEBUG` is generated by CMake source code. This could be different if we change a build system, or if we use a different compiler platform or something.

In any case, I think we should either move everything to `DEBUG` or everything to `NDEBUG`. This is probably out of scope for this PR.

Reply:
> NDEBUG is standard when it exists, but we don't explicitly define it.

Well, that's the point, it's mandated by the C standard to be defined by the standard headers or tools, so it's the same e.g. on different platforms, and we don't have to define it ourselves.

Reply:
It's not mandated to be defined as far as I know. It's just mandated to skip `assert.h` if it exists (in the C standard). There are various discussions about it in the internetz, it's not very clear cut. In any case, I am in favour of using `NDEBUG` everywhere replacing `DEBUG`, but this is probably not the PR to do this.